tsdb/engine/tsm1/encoding.go

```diff
@@ -70,6 +70,10 @@ func (v Values) MaxTime() int64 {
 }
 
 func (v Values) Encode(buf []byte) ([]byte, error) {
+	if len(v) == 0 {
```

Comment:
Yes.  It should not happen based on `tsm1` engine being the caller but it's a public func so some other code could call it at some point.  Personally, I don't a func should make assumptions about how it's called or who is calling it.  I'd rather document that calling `Encode` with an empty slice will panic so it's clear how it behaves.